components/bl-mapbox/src/helpers.js

```diff
@@ -177,51 +177,61 @@ export const initMapboxLibrary = (mapRef, mapContainerRef, component, eventHandl
 };
 
 export const useMarkers = (markers, mapRef, onMarkerClick) => {
-  const [markersArray, setMarkersArray] = useState([]);
+  const markerElements = useRef();
 
   useEffect(() => {
-    if (markers?.length) {
-      markersArray.forEach(marker => {
-        marker.remove();
-      });
-
-      setMarkersArray([]);
+    removeMarkers(markerElements.current);
 
-      markers.forEach(markerItem => {
-        const { color, description } = markerItem;
+    if (markers?.length) {
+      markerElements.current = markers.map(markerItem => {
+        const { color, description, data, coordinates: { lat, lng } } = markerItem;
 
         const marker = new Marker({ color })
-          .setLngLat([markerItem.coordinates.lng, markerItem.coordinates.lat])
+          .setLngLat([lng, lat])
           .addTo(mapRef.current);
 
         const popup = new Popup();
 
         popup.on('open', () => {
-          const coordinates = { lat: markerItem.coordinates.lat, lng: markerItem.coordinates.lng };
+          const coordinates = { lat, lng };
+
+          onMarkerClick({ coordinates, description: description || '', data });
 
-          onMarkerClick({ coordinates, description: description || '' });
+          changeMarkerColor(marker);
         });
 
         if (description) {
           popup.setText(description);
         }
 
-        setMarkersArray(prev => [...prev, marker]);
+        popup.on('close', () => {
+          changeMarkerColor(marker);
+        });
 
         marker.setPopup(popup);
+
+        return marker;
       });
     }
   }, [markers]);
 };
 
+const removeMarkers = markers => {
+  if (markers?.length) {
+    markers.forEach(marker => {
+      marker.remove();
+    });
+  }
+};
+
+const changeMarkerColor = marker => {
+  marker.getElement().childNodes[0].classList.toggle('mapbox-marker-active');
```

Comment:
Not it looks in style like

`bl-customComponent-mapbox mapbox-marker-active`

We don't need to repeat 'mapbox'.

Also if it is possible to remove childNodes from code higher, it could be `marker-root`

Last but not least - active is modification, so better to use another separation(double dash) -> `marker-root--active`